### PR TITLE
Add Lesson 7

### PR DESCRIPTION
**Pull request by EugeniyBurmistrov**

### CODE REVIEW

File: docs/basics/tutorial/lesson-7.md [hunk 1]
```diff
@@ -0,0 +1,286 @@
+## **Lesson 7** - Functionality Enhancements
+
+[\[Get the Code for This Lesson\]](https://github.com/reimagined/resolve/tree/master/examples/shopping-list-tutorial/lesson-7)
+
+This lesson describes additional steps that you can follow to enhance various aspects of the Shopping List application's functionality.
```
Comment (dirk-pieterse) [inline docs/basics/tutorial/lesson-7.md]: This is quite vague.

File: docs/basics/tutorial/lesson-7.md [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+### Modify the Write Side
+
+Define additional events to implement the full set of CRUD (create, read, update, delete) operations.
```
Comment (dirk-pieterse) [inline docs/basics/tutorial/lesson-7.md]: "Define additional events" which events?

File: docs/basics/tutorial/lesson-7.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+```js
+...
+export const SHOPPING_LIST_CREATED = 'SHOPPING_LIST_CREATED'
+export const SHOPPING_LIST_RENAMED = 'SHOPPING_LIST_RENAMED'
+export const SHOPPING_LIST_REMOVED = 'SHOPPING_LIST_REMOVED'
+export const SHOPPING_ITEM_REMOVED = 'SHOPPING_ITEM_REMOVED'
+```
+
+Modify the aggregate projection to enable the shopping list deletion.
```
Comment (dirk-pieterse) [inline docs/basics/tutorial/lesson-7.md]:
```suggestion
Modify the aggregate projection to allow you to delete the shopping list.
```

File: docs/basics/tutorial/lesson-7.md [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+  ) => {
+    await store.update('ShoppingLists', { id: aggregateId }, { $set: { name } })
+  }
+```
+
+### Modify the Frontend
+
+#### Use Static Content
+
+Add the required static content to the application's **static** folder. The example application uses the following static files:
```
Comment (dirk-pieterse) [inline docs/basics/tutorial/lesson-7.md]:
```suggestion
Add the static content to the application's **static** folder. The example application uses the following static files:
```

File: docs/basics/tutorial/lesson-7.md [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+
+- The **Styles.css** file - Contains styles used by the application's client components.
```
Comment (dirk-pieterse) [inline docs/basics/tutorial/lesson-7.md]:
```suggestion
- The **Styles.css** file - Contains styles the application's client components use.
```

File: docs/basics/tutorial/lesson-7.md [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+- The **fontawesome.min.css** file an the **webfonts** folder - The standard [Font Awesome](https://fontawesome.com/) distribution.
+- The **close-button.png** image - An icon that the button used to remove shopping list items displays.
```
Comment (dirk-pieterse) [inline docs/basics/tutorial/lesson-7.md]: An icon that the button used?

File: docs/basics/tutorial/lesson-7.md [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+
+#### Update Components
+
+Modify the ShoppingLists component to provide UI for shopping list deletion.
```
Comment (dirk-pieterse) [inline docs/basics/tutorial/lesson-7.md]: "provide UI"? a UI? the UI?

File: docs/basics/tutorial/lesson-7.md [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+
+```
+
+```js
+const { lists, createShoppingList, removeShoppingList } = this.props
+...
+<ShoppingLists lists={lists} removeShoppingList={removeShoppingList} />
+```
+
+Modify the ShoppingList component to provide the capability to rename shopping lists.
```
Comment (dirk-pieterse) [inline docs/basics/tutorial/lesson-7.md]:
```suggestion
Modify the ShoppingList component to allow users to rename shopping lists.
```

File: docs/basics/tutorial/lesson-7.md [hunk 9]
```diff
@@ -0,0 +1,286 @@
...
+<Image
+  className="example-close-button"
+  src="/close-button.png"
+  onClick={removeShoppingItem.bind(null, aggregateId, {
+    id: todo.id
+  })}
+/>
+```
+
+The code below demonstrates the **Image** component's implementation.
```
Comment (dirk-pieterse) [inline docs/basics/tutorial/lesson-7.md]:
```suggestion
The code below implements the **Image** component.
```

File: docs/basics/tutorial/lesson-7.md [hunk 10]
```diff
@@ -0,0 +1,286 @@
...
+```js
+<Header
+  css={['/fontawesome.min.css', '/style.css', ...]}
+  ...
+/>
+```
+
+#### Update the Optimistic Updates Code
+
+Modify the code related to optimistic UI updates to support the shopping list deletion.
```
Comment (dirk-pieterse) [inline docs/basics/tutorial/lesson-7.md:246]: optimistic?

File: docs/basics/tutorial/lesson-7.md (hunk 2 above)
Comment (EugeniyBurmistrov) [reply docs/basics/tutorial/lesson-7.md]: Changed to 'the following'

File: docs/basics/tutorial/lesson-7.md (hunk 6 above)
Comment (EugeniyBurmistrov) [reply docs/basics/tutorial/lesson-7.md]: Changed to 'The Remove Shopping List button's icon.'

File: docs/basics/tutorial/lesson-7.md (hunk 7 above)
Comment (EugeniyBurmistrov) [reply docs/basics/tutorial/lesson-7.md]: Added 'a'

File: docs/basics/tutorial/lesson-7.md (hunk 10 above)
Comment (EugeniyBurmistrov) [reply docs/basics/tutorial/lesson-7.md:246]: This is a common software design pattern. For example:
https://medium.com/@_erikaybar/optimistic-ui-updates-in-react-9e139ffa2e45
https://www.reddit.com/r/reactjs/comments/3y80w7/what_is_an_optimistic_update/

File: docs/basics/tutorial/lesson-7.md (hunk 1 above)
Comment (EugeniyBurmistrov) [reply docs/basics/tutorial/lesson-7.md]: Changed to: 
"The lesson describes how to perform the following enhancements to the Shopping List application's functionality:

- Modify the reSolve backend to provide the complete set of CRUD (create, read, update, delete) operations.
- Modify the frontend to support all CRUD operations.
- Add static resources to the frontend."

Comment (dirk-pieterse) [reply docs/basics/tutorial/lesson-7.md]: "perform the following enhancements" to the "application's functionality"? How can you perform enhancements to its functionality?

File: docs/basics/tutorial/lesson-7.md (hunk 2 above)
Comment (dirk-pieterse) [reply docs/basics/tutorial/lesson-7.md]: Define the following events to implement the full set of CRUD (create, read, update, delete) operations:

File: docs/basics/tutorial/lesson-7.md (hunk 10 above)
Comment (dirk-pieterse) [reply docs/basics/tutorial/lesson-7.md:246]: Ok

File: docs/basics/tutorial/lesson-7.md (hunk 2 above)
Comment (EugeniyBurmistrov) [reply docs/basics/tutorial/lesson-7.md]: done

File: docs/basics/tutorial/lesson-7.md (hunk 1 above)
Comment (EugeniyBurmistrov) [reply docs/basics/tutorial/lesson-7.md]: 'how to make the following enhancements to the Shopping List application'

Comment (dirk-pieterse) [reply docs/basics/tutorial/lesson-7.md]: how to improve the Shopping List application
OR
how to expand the Shopping List application's functionality

File: docs/basics/tutorial/lesson-7.md (hunk 2 above)
Comment (dirk-pieterse) [reply docs/basics/tutorial/lesson-7.md]: Ok

File: docs/basics/tutorial/lesson-7.md (hunk 6 above)
Comment (dirk-pieterse) [reply docs/basics/tutorial/lesson-7.md]: Ok

File: docs/basics/tutorial/lesson-7.md (hunk 7 above)
Comment (dirk-pieterse) [reply docs/basics/tutorial/lesson-7.md]: Ok

**Pull request merged**